docs/agents/distributed-tracing.md

```diff
@@ -0,0 +1,67 @@
+
+## Http Headers
+
+Our implementation relies on the [W3C Trace Context](https://www.w3.org/TR/trace-context-1/) standard. Until this standard became final,
+we used the header name `elastic-apm-traceparent` and we did not support `tracestate`. Soon after the standard became official, we
+started to fully align with it. For backward compatibility reasons, this was done in phases, so that the first step was to look for both
+`traceparent` headers in incoming requests (meaning - both `elastic-apm-traceparent` and `traceparent`) and sending both `traceparent` headers in outgoing requests (with the exception of the RUM agent,
+due to CORS). [Issue #71](https://github.com/elastic/apm/issues/71) describes this in more detail, as well as tracking implementation
+in the different agents.
+New agents may decide whether to support both `traceparent` headers (so to be compatible with older agent versions) or only the formal W3C
+header.
+
+## Tracestate
+
+For our own `es` `tracestate` entry we will introduce a `key:value` formatted list of attributes.
+This is used to propagate the sampling rate downstream, for example.
+See the [sampling](sampling.md) specification for more details.
+
+The general `tracestate` format is:
+
+    tracestate: es=key:value;key:value...,othervendor=<opaque>
+
+For example:
+
+    tracestate: es=s:0.1,othervendor=<opaque>
+
+
+### Validation and length limits
+
+The [`tracestate`](https://www.w3.org/TR/trace-context/#tracestate-header)
+specification lists a number of validation rules.
+In addition to that,
+there are specific rules for the attributes under the `es` entry.
+
+Agents MUST implement these validation rules when mutating `tracestate`:
+
+- The `tracestate` field may contain a maximum of 32 entries.
+  An entry consists of a vendor key, and an opaque vendor value.
+- Vendor keys (such as `es`) have a maximum size of 256 chars.
+- Vendor keys MUST begin with a lowercase letter or a digit,
+  and can only contain lowercase letters (`a-z`),
+  digits (`0-9`), underscores (`_`), dashes (`-`), asterisks (`*`),
+  and forward slashes (`/`).
+- Vendor values have a maximum size of 256 chars.
+- Vendor values may only contain ASCII RFC0020 characters (i.e., the range `0x20` to `0x7E`) except comma `,` and `=`.
+- In addition to the above limitations, the keys and values used in the `es` entry must not contain the characters `:` and `;`.
+- If adding another key/value pair to the `es` entry would exceed the limit of 256 chars,
+  that key/value pair MUST be ignored by agents.
+  The key/value and entry separators `:` and `;` have to be considered as well.
+
+Note that we will currently only ever populate an `es` `tracestate` entry at the trace root.
+It is not strictly necessary to validate `tracestate` in its entirety when received downstream.
+Instead, downstream agents may opt to only parse the `es` entry skip validation of other vendors' entries.
```

Comment:
Ah, right, we still have to parse the `es` entry because we need to set the `sample_rate` field on transactions and spans, right? It is sort of implied but could you call that out specifically? Maybe create a `## Non-root transactions` section. Or include in one of the existing ones, whatever you think is most appropriate.

Reply:
I've attempted to clarify this in the "Propagation" section in sampling.md. Let me know if it that's good enough, or what's missing